app/src/main/java/com/better/alarm/logger/LoggerModule.kt

```diff
@@ -0,0 +1,131 @@
+package com.better.alarm.logger
+
+import android.content.Context
+import ch.qos.logback.classic.AsyncAppender
+import ch.qos.logback.classic.LoggerContext
+import ch.qos.logback.classic.android.LogcatAppender
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.Appender
+import ch.qos.logback.core.rolling.RollingFileAppender
+import ch.qos.logback.core.rolling.TimeBasedRollingPolicy
+import org.koin.dsl.module
+import org.slf4j.ILoggerFactory
+
+/**
+ * Creates a module which exports a [LoggerFactory] to create loggers. These [Logger]s are backed by
+ * a [RollingFileAppender] and a [LogcatAppender].
+ */
+fun loggerModule() = module {
+  single<LoggerFactory> {
+    object : LoggerFactory {
+      val logback: ILoggerFactory = configureLogback(get())
+      override fun createLogger(tag: String): Logger {
+        return Logger(logback.getLogger(tag))
+      }
+    }
+  }
+}
+
+private fun configureLogback(context: Context): ILoggerFactory {
```

Comment:
@kolanse this is the configuration for Logback, adding 2 appenders: LogCat and rolling file with 3 history items. `isCleanHistoryOnStart` doesn't slow the startup much so I chose to leave it.

Reply:
Wow ... i really learnt a lot from the code and its implementation. Thank you very much for the tag!!!!